config/v1/types_authentication.go

```diff
@@ -222,11 +223,21 @@ type OIDCProvider struct {
 	//
 	// +listType=atomic
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// UserValidationRules defines the set of rules used to validate claims in a user’s token.
+	// These rules determine whether a token subject is considered valid based on its claims.
+	// Each rule is evaluated independently.
+	// See the TokenUserValidationRule type for more information on rule structure.
+	// +optional
+
+	//+listType=atomic
+	UserValidationRules []TokenUserValidationRule `json:"userValidationRules,omitempty"`
 }
 
 // +kubebuilder:validation:MinLength=1
 type TokenAudience string
 
+// +kubebuilder:validation:XValidation:rule="self.discoveryURL.size() > 0 ? (self.url.size() == 0 || self.discoveryURL.find('^.+[^/]') != self.url.find('^.+[^/]')) : true",message="discoveryURL must be different from URL"
```

Comment:
I believe that `self.url` needs to be `self.issuerURL` - the CEL expressions will go based on the serialized field name:
```suggestion
// +kubebuilder:validation:XValidation:rule="self.discoveryURL.size() > 0 ? (self.issuerURL.size() == 0 || self.discoveryURL.find('^.+[^/]') != self.issuerURL.find('^.+[^/]')) : true",message="discoveryURL must be different from issuerURL"
```

Additionally, you want this to be a feature-gated validation using `// +openshift:validation:FeatureGateAwareXValidation:featureGate=ExternalOIDCWithNewAuthConfigFields,rule=....`. If this isn't feature-gated, it will attempt to be applied by default and will fail compilation because there would be no such field as `discoveryURL`.